sdk/tests/core/test_az_url_encode.c

```diff
@@ -84,6 +85,10 @@ static void test_url_encode_basic(void** state)
     assert_true(az_span_is_content_equal(
         AZ_SPAN_FROM_BUFFER(buf20), AZ_SPAN_FROM_STR("aBc%2Fg*************")));
   }
+  {
+    int32_t url_length = _az_span_url_encode_calc_length(AZ_SPAN_FROM_STR("aBc/g"));
+    assert_int_equal(url_length, sizeof("aBc%2Fg") - 1);
```

Comment:
Add more test cases? I would consider adding a call to `_az_span_url_encode_calc_length` and asserting the expected length for all the test strings we call `_az_span_url_encode` to in the existing tests.

Cases such as:
- empty span
- nothing is encoded
- first/last character is encoded
- every character is encoded
- % character, which in itself is encoded as %25